### PR TITLE
fix(csp): re-unify the Content-Security-Policy headers

### PR DESCRIPTION
**Pull request by RomainMuller**
Stop serving the `Clear-Site-Data` header (as if honored, it would
result in clearing local storage, which is user-hostile as it'll nuke
the user's preferred language, etc...).

Re-unify the CSP-setting CloudFront function so that a consistent CSP is
served, instead of not serving one (or serving an incorrect one) for
select pages.

The incorrect/questionable behavior was introduced in #515, but is likely
causing issues on browsers such as Safari or Chrome.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: src/webapp/index.ts [hunk 1]
```diff
@@ -209,7 +196,7 @@ export class WebApp extends Construct {
     }
 
     const websiteOrigin = new origins.S3Origin(this.bucket);
-    this.distribution.addBehavior('/index.html', websiteOrigin, indexBehaviorOptions);
+    this.distribution.addBehavior('/*', websiteOrigin, behaviorOptions);
```
Comment (iliapolo) [inline src/webapp/index.ts:199]: So this means the now everything will have caching disabled - was this the intent? The PR talks about unification of headers, not changes in cache behavior.

Comment (RomainMuller) [reply src/webapp/index.ts:199]: Caching was disabled before this change...?

Comment (iliapolo) [reply src/webapp/index.ts]: Right, missed that 👍

File: src/webapp/index.ts [hunk 2]
```diff
@@ -160,11 +159,10 @@ export class WebApp extends Construct {
     // it is changed the function will be recreated.
     // see https://github.com/aws/aws-cdk/issues/15523
     const functionId = `AddHeadersFunction${this.node.addr}`;
-    const indexFunctionId = `IndexHeadersFunction${this.node.addr}`;
 
     const behaviorOptions: cloudfront.AddBehaviorOptions = {
       compress: true,
-      cachePolicy: cloudfront.CachePolicy.CACHING_DISABLED,
+      cachePolicy: cloudfront.CachePolicy.CACHING_OPTIMIZED,
```
Comment (iliapolo) [inline src/webapp/index.ts:165]: So why this change then? You mentioned caching was disabled, but is now re-enabled with `max-age` 0? I'm not sure I follow the reasoning here.

Comment (RomainMuller) [reply src/webapp/index.ts:165]: With the hubris around latencies I felt like re-enabling cache was desired... now I don't know what we want to do anymore.

Comment (iliapolo) [reply src/webapp/index.ts:165]: I suggest to scope this PR down to only handle the cookie banner problems. I assume the caching policies are unrelated to the right? 

We can discuss caching in a subsequent PR.

**Review by iliapolo — APPROVED** [review]
